tests/expr_and_series/list/len_test.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import pytest
+
+import narwhals.stable.v1 as nw
+from tests.utils import PANDAS_VERSION
+from tests.utils import Constructor
+from tests.utils import ConstructorEager
+from tests.utils import assert_equal_data
+
+data = {"a": [[1, 2], [3, 4, None], None, [], [None]]}
+expected = {"a": [2, 3, None, 0, 1]}
+
+
+def test_len_expr(
+    request: pytest.FixtureRequest,
+    constructor: Constructor,
+) -> None:
+    if any(backend in str(constructor) for backend in ("dask", "modin", "cudf")):
+        request.applymarker(pytest.mark.xfail)
+
+    if "pandas" in str(constructor) and PANDAS_VERSION < (2, 2):
+        request.applymarker(pytest.mark.xfail)
+
+    result = nw.from_native(constructor(data)).select(
+        a=nw.col("a").cast(nw.List(nw.Int32())).list.len()
```

Comment:
This seems like a minor detail, yet it is a big deal, because pandas seems to not preserve the series name. I opened an issue upstream [pandas#60522](https://github.com/pandas-dev/pandas/issues/60522).

I tried a workaround with forcing `root_names` and `output_names` in the `ListNamespace` method but I couldn't manage to make it work properly.